File: backend/app/routers/methodology.py
from __future__ import annotations

from fastapi import APIRouter

from ..config import (
    DISCLAIMER,
    ENCODING_SCALE_BIN,
    EXPOSURE_PRESETS,
    FITZPATRICK,
    H_D_MAX,
    K_MINUTES,
    MODEL_VERSION,
    TEMP_ENCODING_SCALE_BIN,
    TEMP_OFFSET,
)
from ..models import MethodologyResponse

router = APIRouter(prefix="/api", tags=["methodology"])

EQUATIONS: dict[str, str] = {
    "t_minutes": "(K_minutes * k_skin) / (H_D_month * f_cover)",
    "infinity_rule": "H_D_month <= 0 OR f_cover <= 0 → Infinity",
}


@router.get("/methodology", response_model=MethodologyResponse)
def methodology() -> MethodologyResponse:
    return MethodologyResponse(
        model_version=MODEL_VERSION,
        equations=EQUATIONS,
        constants={
            "K_minutes": K_MINUTES,
        },
        fitzpatrick_table=FITZPATRICK,
        exposure_presets=EXPOSURE_PRESETS,
        encoding={
            "scale": ENCODING_SCALE_BIN,
            "H_D_max": H_D_MAX,
            "temp_encoding_scale": TEMP_ENCODING_SCALE_BIN,
            "temp_offset": TEMP_OFFSET,
        },
        disclaimer=DISCLAIMER,
    )

File: src/api/methodology.ts
import type { MethodologyResponse } from "../types";

export const METHODOLOGY: MethodologyResponse = {
  model_version: "2.0.0",
  equations: {
    t_minutes: "(K_minutes * k_skin) / (H_D_month * f_cover)",
    infinity_rule: "H_D_month <= 0 OR f_cover <= 0 → Infinity",
  },
  constants: {
    K_minutes: 20.2,
  },
  fitzpatrick_table: {
    "1": 1.0,
    "2": 1.2,
    "3": 1.5,
    "4": 2.0,
    "5": 2.8,
    "6": 3.8,
  },
  exposure_presets: {
    face_hands: 0.05,
    tshirt_shorts: 0.25,
    swimsuit: 0.85,
  },
  encoding: {
    scale: 3,
    H_D_max: 20_000,
    temp_encoding_scale: 100,
    temp_offset: 50,
  },
  disclaimer:
    "This is an EDUCATIONAL MODEL. " +
    "It is NOT medical advice. " +
    "It does NOT diagnose vitamin D deficiency.",
};
